github/internal/retry.go

```diff
@@ -86,6 +89,20 @@ func GithubErrChecker(maxTries int, sleeper func(time.Duration)) func(error) boo
 			return true
 		}
 
+		if strings.Contains(err.Error(), "secondary rate limit. Please wait") {
+			rtime, err := rand.Int(rand.Reader, big.NewInt(30))
+			if err != nil {
+				logrus.Error(err)
+				return false
+			}
+			waitDuration := time.Duration(rtime.Int64()*int64(time.Second)) + defaultGithubSleep
+			logrus.
+				WithField("err", err).
+				Infof("Hit the GitHub secondary rate limit on try %d, sleeping for %s", try, waitDuration)
+			sleeper(waitDuration)
+			return true
+		}
```

Comment:
How about our idea of locking the gatherer threads if a rate limit occurs? Would that probably prevent the secondary rate limit?

Ref https://docs.github.com/en/rest/overview/resources-in-the-rest-api#secondary-rate-limits